src/analysis/SystemMonitor.hpp

```diff
@@ -35,15 +35,44 @@
 #include "NPart.hpp"
 #include "storage/Storage.hpp"
 #include "iterator/CellListIterator.hpp"
+#include <boost/mpi/timer.hpp>
 
 namespace espressopp
 {
 namespace analysis
 {
-class SystemMonitorOutputCSV
+/**
+ * Base class for SystemMonitor output formats.
+ */
+class SystemMonitorOutput
 {
     friend class SystemMonitor;
 
+public:
+    virtual void write() = 0;
+    void setSystem(shared_ptr<System> system) { system_ = system; }
```

Comment:
Not really important, more matter of style. This will introduce an unnecessary copy, you could use const&.

Reply:
I think here we should copy the shared_ptr to share the ownership.

Reply:
Yes, of course, keep the shared_ptr. I meant you are copying the shared_ptr twice. Not relevant for a shared_ptr but relevant if you copy something like std::vector or std::string.

Reply:
So it's better to use:
```suggestion
    void setSystem(const shared_ptr<System> &system) { system_ = system; }
```
?